__tests__/with-urql-client.spec.tsx

```diff
@@ -72,14 +70,14 @@ describe('withUrqlClient', () => {
   });
 
   describe('with ctx callback to create client options', () => {
-    let Component: NextFC<any>;
+    let Component: NextComponentType<any>;
 
     // Fake up a string to simulate, say, a token accessed via browser cookies or localStorage.
     const token = Math.random()
       .toString(36)
       .slice(-10);
 
-    const mockContext: NextContextWithAppTree = {
+    const mockContext: NextUrqlContext = {
       AppTree: MockAppTree,
       pathname: '/',
       query: {
```

Comment:
FYI I'm also seeing a TS error on L87 (in the new diff):

```
Type '{ headers: { cookie: string; }; }' is missing the following properties from type 'IncomingMessage': httpVersion, httpVersionMajor, httpVersionMinor, complete, and 40 more.
```

We'll just have to coerce the `req` type like so:

```
req: {
  headers: {
    cookie: token,
  },
} as NextUrqlContext['req']
```

Reply:
Oh cool, didn't know about accessing a type property like that!